src/components/ComponentTree.js

```diff
@@ -0,0 +1,53 @@
+import React, {useState, useEffect} from "react";
+
+const counterKey = "counter";
+
+function ComponentTree() {
+
+    const [count, setCount] = useState(0);
+    useEffect(() => {
+        if (count === 0) {
+            setCount(parseInt(localStorage.getItem(counterKey)))
+        } else {
+            localStorage.setItem(counterKey, count)
+        }
+    }, [count]);
+
+    return (
+        <>
+            <Component2 setCount={setCount} count={count}></Component2>
+            <Component3 count={count}></Component3>
+        </>
+    );
+}
+
+function Component2(props) {

```

Comment:
Doesn't matter too much as there's just one thing getting passed through - but good to commonize and add a type to these props